clients/src/main/java/org/apache/kafka/common/utils/Checksums.java

```diff
@@ -29,22 +32,72 @@
  * NOTE: This class is intended for INTERNAL usage only within Kafka.
  */
 public final class Checksums {
+    private static final MethodHandle BYTE_BUFFER_UPDATE;
+
+    static {
+        MethodHandle byteBufferUpdate = null;
+        if (Java.IS_JAVA9_COMPATIBLE) {
+            try {
+                byteBufferUpdate = MethodHandles.publicLookup().findVirtual(Checksum.class, "update",
+                    MethodType.methodType(void.class, ByteBuffer.class));
+            } catch (Throwable t) {
+                handleUpdateThrowable(t);
```

Comment:
I changed this to throw an exception so that our tests fail if the optimization stops working due to a bug.